File: C Programs ☠️/countNodes.c
// Write a function to count the number of node present in a linked list.

#include <stdio.h>
#include <stdlib.h>

typedef struct node
{
    int data;
    struct node *next;
} NODE;

NODE *createList(NODE *);
int countNodes(NODE *);

int main()
{
    int n;
    NODE *head = NULL;
    head = createList(head);
    n = countNodes(head);
    printf("Number Of Nodes in this Linked List is %d", n);
    return 0;
}

NODE *createList(NODE *head)
{
    NODE *newNode, *ptr;
    int num;
    printf("Creating Linked List...\n");
    printf("Enter -1 to end\n");
    printf("Enter the data: ");
    scanf("%d", &num);
    while (num != -1)
    {
        newNode = (NODE *)malloc(sizeof(NODE));
        newNode->data = num;
        if (head == NULL)
        {
            newNode->next = NULL;
            head = newNode;
        }
        else
        {
            ptr = head;
            while (ptr->next != NULL)
            {
                ptr = ptr->next;
            }
            ptr->next = newNode;
            newNode->next = NULL;
        }
        printf("Enter the data: ");
        scanf("%d", &num);
    }
    return head;
}

int countNodes(NODE *head)
{
    int count = 0;
    NODE *ptr = head;
    while (ptr != NULL)
    {
        count++;
        ptr = ptr->next;
    }
    return count;
}

// OUTPUT
// Creating Linked List...
// Enter -1 to end
// Enter the data: 5
// Enter the data: 6
// Enter the data: 7
// Enter the data: 8
// Enter the data: -1
// Number Of Nodes in this Linked List is 4

File: Linked List/Python Programs 🐍/Singly/countNodes.py
# Write a function to count the number
# of node present in a linked list.

class Node:
    def __init__(self, data):
        self.data = data
        self.next = None


class LinkedList:
    def __init__(self):
        self.head = None

    def createList(self):
        print("Creating Linked List...")
        print("Enter -1 to end")
        num = int(input("Enter the data: "))
        while(num != -1):
            newNode = Node(num)
            if self.head is None:
                self.head = newNode
            else:
                last = self.head
                while last.next:
                    last = last.next
                last.next = newNode
            num = int(input("Enter the data: "))

    def size(self):
        count = 0
        temp = self.head
        while temp:
            count += 1
            temp = temp.next
        return count


myList = LinkedList()
myList.createList()
n = myList.size()
print("Number of Nodes is: " + str(n))


# OUTPUT
# Creating Linked List...
# Enter -1 to end
# Enter the data: 6
# Enter the data: 3
# Enter the data: 8
# Enter the data: 2
# Enter the data: -1
# Number of Nodes is: 4
